app/models/concerns/job.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Job
+  extend ActiveSupport::Concern
+
+  STATUSES = %w[queued cancelled running completed errored].freeze
```

Comment:
Im not sure if this is still a thing in rails... but I remmeber if the items in the array change its order it will cause some wierd issues because its been stored in the db as its index position.

so I'd either add a spec to ensure it has the correct order, or update it so use a hash:
https://naturaily.com/blog/ruby-on-rails-enum

Reply:
nice! I'll make it a hash. Thanks bro :)